backend/data/generate_flight_data.py

```diff
@@ -0,0 +1,93 @@
+import os
+import json
+import random
+from datetime import datetime, timedelta
+
+def generate_flight_data(start_date_str):
+    # Convert input string to datetime
+    start_date = datetime.strptime(start_date_str, "%d-%m-%Y %H:%M:%S")
+
+    # List of airports
+    airports = ["SFO", "LAX", "NYC", "CHI", "MIA", "DFW", "SEA", "BOS"]
+
+    # List of airlines with code, name, and flight number range
+    airlines = [
+        {"code": "AA", "name": "American Airlines", "flight_range": (1000, 2000)},
+        {"code": "DL", "name": "Delta Air Lines", "flight_range": (2000, 3000)},
+        {"code": "UA", "name": "United Airlines", "flight_range": (3000, 4000)},
+        {"code": "WN", "name": "Southwest Airlines", "flight_range": (4000, 5000)},
+        {"code": "B6", "name": "JetBlue Airways", "flight_range": (5000, 6000)},
+        {"code": "AS", "name": "Alaska Airlines", "flight_range": (6000, 7000)},
+        {"code": "NK", "name": "Spirit Airlines", "flight_range": (7000, 8000)},
+        {"code": "F9", "name": "Frontier Airlines", "flight_range": (8000, 9000)}
+    ]
+
+    # Approximate distances between airports (in miles)
+    distances = {
+        ("SFO", "LAX"): 337, ("SFO", "NYC"): 2572, ("SFO", "CHI"): 1847, ("SFO", "MIA"): 2582, ("SFO", "DFW"): 1464, ("SFO", "SEA"): 679, ("SFO", "BOS"): 2704,
+        ("LAX", "NYC"): 2475, ("LAX", "CHI"): 1745, ("LAX", "MIA"): 2342, ("LAX", "DFW"): 1235, ("LAX", "SEA"): 954, ("LAX", "BOS"): 2611,
+        ("NYC", "CHI"): 740, ("NYC", "MIA"): 1090, ("NYC", "DFW"): 1372, ("NYC", "SEA"): 2414, ("NYC", "BOS"): 215,
+        ("CHI", "MIA"): 1197, ("CHI", "DFW"): 802, ("CHI", "SEA"): 1721, ("CHI", "BOS"): 867,
+        ("MIA", "DFW"): 1121, ("MIA", "SEA"): 2724, ("MIA", "BOS"): 1258,
+        ("DFW", "SEA"): 1660, ("DFW", "BOS"): 1551,
+        ("SEA", "BOS"): 2489
+    }
+
+    # Generate flight data
+    flight_data = {}
+
+    for airport in airports:
+        flight_data[airport] = {}
+        # Randomly select 3-4 destination airports
+        destinations = random.sample([a for a in airports if a != airport], random.randint(3, 4))
+        
+        for dest in destinations:
+            flight_data[airport][dest] = []
+            
+            # Generate up to 8 flights for each destination
+            num_flights = random.randint(4, 8)
```

Comment:
I think 8 flights per day would be better, otherwise you might not have enough flights within 24h.

Reply:
Also, it looks like departure time will be 1 each day, right? Generated data won't generate multiple flights within the same day with this code:

`dep_time = start_date + timedelta(days=random.uniform(0, 7))`

So we might want to do something like this:

```
num_days = 7 

for _ in range(num_days):

for _ in range(num_flights)

dep_time = start_date + timedelta(hours=random.uniform(0, 23))
```